netbout-web/src/test/java/com/netbout/rest/bout/TkAttachTest.java

```diff
@@ -137,6 +153,57 @@ public void ignoresWrongRequest() throws Exception {
     }
 
     /**
+     * TkAttach can ignore wrong file without saving it.
+     * @throws Exception If there is some problem inside
+     */
+    @Test
+    public void ignoresWrongFile() throws Exception {
+        final MkBase base = new MkBase();
+        final String urn = "urn:test:2";
+        final User user = base.user(new URN(urn));
+        user.aliases().add("jeff2");
+        final Alias alias = user.aliases().iterate().iterator().next();
+        final long bout = alias.inbox().start();
+        alias.inbox().bout(bout).friends().invite(alias.name());
+        final String file = "test.bin";
+        final RqWithAuth request = new RqWithAuth(
+            urn,
+            new RqMultipart.Fake(
+                TkAttachTest.fake(bout),
+                new RqWithHeaders(
+                    new Request() {
+                        @Override
+                        public Iterable<String> head() throws IOException {
+                            return new LinkedList<>();
+                        }
+                        @Override
+                        public InputStream body() throws IOException {
+                            return new ByteArrayInputStream("".getBytes());
+                        }
+                    },
+                    String.format(TkAttachTest.POST_URL, bout),
+                    //@checkstyle LineLengthCheck (1 line)
+                    String.format("Content-Disposition: form-data; name=\"file\"; filename=\"%s\"", file),
+                    "Content-Type: application/octet-stream"
+                )
+            )
+        );
+        try {
+            new FkBout(".*$", new TkAttach(base)).route(request);
+        } catch (final RsFailure ex) {
+            MatcherAssert.assertThat(
+                "file unexpectedly exists after attach failure",
+                new Attachments.Search(
+                    alias.inbox().bout(bout).attachments()
+                ).exists(file),
+                Matchers.is(false)
+            );
+            return;
+        }
+        Assert.fail("Expected RsFailure exception but nothing was thrown");
```

Comment:
@exper0 let's move this line before `catch`
and remove `return` in catch clause